### PR TITLE
Justine/persisted repository

### LINKED ISSUE #21: Refactor RemoveUserRoom from PersistedRepository
Figure out a good replacement in EF Core

```

public void RemoveUserRoom(ChatUser user, ChatRoom room)
{
    RunNonLazy(() =>
    {
        // The hack from hell to attach the user to room.Users so delete is tracked
        ObjectContext context = ((IObjectContextAdapter)_db).ObjectContext;
        RelationshipManager manger = context.ObjectStateManager.GetRelationshipManager(room);
        IRelatedEnd end = manger.GetRelatedEnd("JabbR.Models.ChatRoom_Users", "ChatRoom_Users_Target");
        end.Attach(user);

        room.Users.Remove(user);
    });
}
```

### PR DESCRIPTION
**Pull request by jcocchi**
Merging the new models, changing Jabbr-Core code to use the new models, database migrations, and testing. Fixes #21 #25 #66 #67 #93 #100 #121 #132

### CODE REVIEW

**Comment by adamtuliper** [comment]
Need :shipit: - anyone?

**Comment by heatherbshapiro** [comment]
I tried to run the tests both ways and am getting errors. 

When I run from test explorer, the project builds and I get the error
`Failed to initialize client proxy: could not connect to test process vstest.discoveryengine.x86.exe.`
-- I'm not sure where this error is coming from. Any ideas?

When I run dotnet test against Jabbr-Core.Tests I am getting the following error: 
```
dotnet test
dotnet : dotnet-test Error: 0 : System.InvalidOperationException: C:\Users\heathes\Documents\Mach 
Upskilling\JabbR-Core\project.json does not exist.
At line:1 char:1
+ dotnet test
+ ~~~~~~~~~~~
```

It seems that it is looking for the file in the wrong place. The location should be : 
C:\Users\heathes\Documents\Mach Upskilling\JabbR-Core\src\Jabbr-Core.Tests\project.json

When I cd'ed into this folder in the console, I was able to get the tests to run but have 21 errors.

**Comment by KSLHacks** [comment]
I'm going to try to duplicate Heather's errors and see if its a local machine/branch error. Two testers might help in this case.

**Comment by KSLHacks** [comment]
@jcocchi  very clear test directions and expectations.

- I was able to checkout the branch justine/persistedRepository and follow the testing instructions.= 
- Project compiles successfully
- Expected functionality (Load Lobby with no rooms, create room, add messages)

:shipit: 

image

**Comment by KSLHacks** [comment]
Was not able to replicate @heatherbshapiro 's errors.

**Comment by heatherbshapiro** [comment]
So I had to run Visual Studio as administrator, reinstall xunit in the package manager for Jabbr-Core.Data, and now it is at least viewing the tests in the test explorer. (Really not sure why)

I redid the whole process and at first I got 11 errors, but then reran and got 10. Reran again and got a different 10th error. Can see my error list below. I think we should have a 3rd tester just to be sure.

image


image

image

File: src/JabbR-Core.Data/Models/UserRoomAllowed.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 namespace JabbR_Core.Data.Models
 {
-    public partial class ChatRoomChatUserOwner
+    public partial class UserRoomAllowed
```
Comment (adamtuliper) [inline src/JabbR-Core.Data/Models/UserRoomAllowed.cs]: Looking at PR, weren’t we going to use the names ChatRoomChatUserAllowed and ChatRoomChatUserOwner (so its more in line with what EF shows)? If we’re going to rename away from the EF many-many style (no prob with that), may I suggest we simplify but with intent (does this represent those tables appropriately?)
ChatRoomUsers
ChatRoomOwners

Comment (jcocchi) [inline src/JabbR-Core.Data/Models/UserRoomAllowed.cs]: This is done now, just waiting for Gabby to finish testing

**Comment by adamtuliper** [comment]
Is this all good to go with latest after EF issue resolution (by rolling back version)?

**Comment by gcrev93** [comment]
image

Followed the directions
Results were as expected
I was able to run the code after and execute creating a room and sending messages

**Comment by gcrev93** [comment]
:shipit:

**Pull request merged**